dali/pipeline/operator/builtin/external_source.cc

```diff
@@ -19,11 +19,20 @@ namespace dali {
 template<>
 void ExternalSource<CPUBackend>::RunImpl(HostWorkspace &ws) {
   std::list<uptr_tl_type> tensor_list_elm;
+  std::list<uptr_cuda_event_type> internal_copy_to_storage;
   {
     std::unique_lock<std::mutex> busy_lock(busy_m_);
-    cv_.wait(busy_lock, [&data = tl_data_]{return !data.IsEmpty();});
+    cv_.wait(busy_lock, [&data = tl_data_, &blocking = blocking_] {
```

Comment:
I think it's better to capture the `blocking_` as copy instead of reference as it wouldn't change.

Reply:
Dona